core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -957,3 +980,17 @@ private[spark] class PythonParallelizeServer(sc: SparkContext, parallelism: Int)
   }
 }
 
+/**
+ * Create a socket server class and run user function on the socket in a background thread.
+ * This is the same as calling SocketAuthServer.setupOneConnectionServer except it creates
```

Comment:
I don't think there was a specific reason about the timeout. Cc @vanzin and @squito

Reply:
BTW, there are multiple pleases for such hardcoded timeout. I suspect it won't be a major issue.

Reply:
Oh, the question was why it was not set? It still sets:

https://github.com/apache/spark/blob/branch-2.4/core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala#L852

https://github.com/apache/spark/blob/branch-2.4/core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala#L899

Reply:
FYI. @HyukjinKwon and @cloud-fan also found the timeout value only affects `accept()`

Reply:
Hi, @gatorsmile . Do you mean it affects the on-going `2.4.4` vote?

Reply:
sorry I think I'm a bit lost in the discussion -- @gatorsmile do you think something is wrong here or not?  seems OK to me, the timeout is the same as before, just in a different spot.

I don't think the timeout is crucial for correctness, its more about getting sane errors if there is some bug and nothing connects back.  Rather than having things block forever, you'll get a timeout exception.

Reply:
IIUC the timeout is for establishing the socket connection. Given we build local socket connection between JVM and Python, 10 seconds is fine.

And agree with @squito it's nothing about correctness, if weird thing happens users will get an error instead of wrong result.

Reply:
Yea, it's fine. I don't think there's any major issue with that.